climakitae/threshold_tools.py

```diff
@@ -0,0 +1,613 @@
+
+    distrs = ["gev", "gumbel", "weibull", "pearson3", "genpareto"]
+    if distr not in distrs:
+        raise ValueError(
+            "invalid distr type. expected one of the following: %s" % distrs
+        )
+
+    ams_attributes = ams.attrs
+
+    if multiple_points == True:
+        ams = ams.stack(allpoints=["x", "y"]).squeeze().groupby("allpoints")
+
+    def apply_function(ams):
+        def function(ams):
+
+            if distr == "gev":
+                lmoments = ldistr.gev.lmom_fit(ams)
+
+            if distr == "gumbel":
+                lmoments = ldistr.gum.lmom_fit(ams)
+
+            if distr == "weibull":
+                lmoments = ldistr.wei.lmom_fit(ams)
+
+            if distr == "pearson3":
+                lmoments = ldistr.pe3.lmom_fit(ams)
+
+            if distr == "genpareto":
+                lmoments = ldistr.gpa.lmom_fit(ams)
+
+            return lmoments
+
+        return xr.apply_ufunc(
+            function,
+            ams,
+            input_core_dims=[["time"]],
+            exclude_dims=set(("time",)),
+            output_core_dims=[[]],
+        )
+
+    lmoments = apply_function(ams)
+
+    new_ds = lmoments.to_dataset()
+    new_ds = new_ds.rename({"T2": "lmoments"})
+
+    if multiple_points == True:
+        new_ds = new_ds.unstack("allpoints")
+
+    new_ds.attrs = ams_attributes
+    new_ds.attrs["distribution"] = "{}".format(str(distr))
+
+    return new_ds
+
+
+#####################################################################
+
+
+def get_aicc_stat(ams, multiple_points=True):
+
+    ams_attributes = ams.attrs
+
+    if multiple_points == True:
+        ams = ams.stack(allpoints=["x", "y"]).squeeze().groupby("allpoints")
+
+    def apply_function(ams):
+        def function(ams):
+
+            try:
+                lmoments_gev = ldistr.gev.lmom_fit(ams)
+                aicc_gev = ["gev", lstats.AICc(ams, "gev", lmoments_gev)]
+            except ValueError:
+                aicc_gev = ["gev", np.nan]
+
+            try:
+                lmoments_gum = ldistr.gum.lmom_fit(ams)
+                aicc_gum = ["gumbel", lstats.AICc(ams, "gum", lmoments_gum)]
+            except ValueError:
+                aicc_gum = ["gumbel", np.nan]
+
+            try:
+                lmoments_wei = ldistr.wei.lmom_fit(ams)
+                aicc_wei = ["weibull", lstats.AICc(ams, "wei", lmoments_wei)]
+            except ValueError:
+                aicc_wei = ["weibull", np.nan]
+
+            try:
+                lmoments_pe3 = ldistr.pe3.lmom_fit(ams)
+                aicc_pe3 = ["pearson3", lstats.AICc(ams, "pe3", lmoments_pe3)]
+            except ValueError:
+                aicc_pe3 = ["pearson3", np.nan]
+
+            try:
+                lmoments_gpa = ldistr.gpa.lmom_fit(ams)
+                aicc_gpa = ["genpareto", lstats.AICc(ams, "gpa", lmoments_gpa)]
+            except ValueError:
+                aicc_gpa = ["genpareto", np.nan]
+
+            all_aicc_results = (
```

Comment:
Instead of a string of all the AICc values which is difficult to use afterward, you could add a coordinate to the `DataArray` with the distribution names and corresponding AICc values. That would also allow you to call the `DataArray.min()` method to find the distribution with the lowest AICc.